### ISSUE TITLE
dylibs produced by a dependency's build script not found by "cargo run"

### ISSUE THREAD

**Issue by stacktracer** [rust-lang/cargo#2765]
I am new to Rust, and trying to get rolling with [libui-rs](https://github.com/pcwalton/libui-rs). My project ("myproj") depends on the "ui" crate, which in turn depends on "ui-sys".

Doing ```cargo build``` in myproj works fine, including the creation of libui.so.0 (built by ui-sys), which ends up in a subdir under myproj/target/. However, ```cargo run``` can't find libui.so.0, and dies with the error message:
```
error while loading shared libraries: libui.so.0: cannot open shared object file: No such file or directory
```

Based on https://github.com/rust-lang/cargo/commit/03eaa91 and https://github.com/rust-lang/cargo/issues/915, it seems that the intention is for ```cargo run``` to set LD_LIBRARY_PATH.

Here is a self-contained example: https://github.com/stacktracer/libui-fiddling. (Note that libui-rs is pulled in as a git submodule -- follow the "Setup" steps in the libui-fiddling readme.)

**Comment by stacktracer** [comment]
This is with the current stable versions:
```
$ rustc --version && cargo --version
rustc 1.9.0 (e4e8b6668 2016-05-18)
cargo 0.10.0-nightly (10ddd7d 2016-04-08)
```

**Comment by mbrubeck** [comment]
When I run this, the following string is ending up in `$LD_LIBRARY_PATH`:

`native=/home/mbrubeck/src/test/libui-fiddling/myproj/target/debug/build/ui-sys-f26fc98857234bd7/out`

The `native=` part should be stripped out, but isn't.

**Comment by stacktracer** [comment]
How are you checking LD_LIBRARY_PATH? If I call ```std::env::var("LD_LIBRARY_PATH")``` in main, I get ```Err(NotPresent)```.

**Comment by mbrubeck** [comment]
There is a fix for this in #2763.
